src/app/firebase_app.ts

```diff
@@ -474,36 +479,30 @@ export function createFirebaseNamespace(): FirebaseNamespace {
     if (factories[name]) {
       error('duplicate-service', {'name': name});
     }
-    if (!!allowMultipleInstances) {
-      // Check if the service allows multiple instances per app
-      factories[name] = createService;
-    } else {
-      // If not, always return the same instance when a service is instantiated
-      // with an instanceString different than the default.
-      factories[name] =
-          (app: FirebaseApp, extendApp?: (props: {[prop: string]: any}) => void,
-           instanceString?: string) => {
-            // If a new instance is requested for a service that does not allow
-            // multiple instances, return the default instance
-            return createService(app, extendApp, DEFAULT_ENTRY_NAME);
-          };
-    }
+
+    /**
+     * If multiple instances are allowed, return the true create service
+     * otherwise, return a proxied reference to the same service
+     */
+    factories[name] = allowMultipleInstances ? createService :
+      (...args) => createService.apply(this, args);
+    
+    // Capture the appHook, if passed
     if (appHook) {
       appHooks[name] = appHook;
+      getApps().forEach(app => {
+        appHook('create', app);
```

Comment:
The two functions that currently will trigger the `create` hook for services are `initializeApp` and `registerService`. Though they are similar they are slightly different and actual have no overlap in terms of when appHooks are fired.

- `initializeApp` attempts to run appHooks for every factory currently in existence on the initialized app.
- `registerService` runs only the new appHook (if applicable) on all existing instances of `FirebaseApp` it then also registers itself as one of the available appHooks so that future `initializeApp` calls will trigger the hook.

The `create` hook will be run once, and only once, per instance of Firebase App. Because one can't register multiple services of the same name, attempting to register multiple appHooks under the same name, and trigger a duplicate function call, should fail with an error long before the call is ever attempted.